src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -471,6 +444,82 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
+func redpandaEnvVars() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "REDPANDA_ENVIRONMENT",
+			Value: "kubernetes",
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.name",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "status.podIP",
+				},
+			},
+		},
+	}
+}
+
+func (r *StatefulSetResource) wasmContainer() corev1.Container {
+	if r.pandaCluster.Spec.DeepCopy().ResourcesWASM == nil {
+		r.logger.Info("BUG! No resources found for wasm - this should never happen with defaulting webhook enabled")
```

Comment:
If this is a `BUG!` it should be at the error log level. right?

Reply:
Maybe message to the user should mention `please consider enabling defaulting webhook`?

Reply:
it could be bug or disabled webhook, also error requires error and I had nothing so I used info :)